docs/extension-development/adding-custom-products-to-add-products-onboarding-list.md

```diff
@@ -0,0 +1,76 @@
+---
+post_title: How to add custom product types to Add Products onboarding list
+menu_title: Add custom product types to Add Products onboarding list
+tags: how-to
+---
+
+## Introduction
+
+WooCommerce allows developers to extend the product type onboarding list, offering a more customizable and engaging experience during the Add Products onboarding task. This tutorial will guide you through adding custom product types to your WooCommerce store using the `experimental_woocommerce_tasklist_product_types` JavaScript filter.
+
+## Prerequisites
+
+- A basic understanding of JavaScript and PHP.
+- WooCommerce 8.8 or later installed on your WordPress site.
+
+## Step 1: Adding a JavaScript Filter
+
+To add a new product type to the onboarding list, we'll utilize the `@wordpress/hooks` package, specifically the addFilter function. If you're not already familiar, `@wordpress/hooks` allows you to modify or extend features within the WordPress and WooCommerce ecosystem without altering the core code.
+
+First, ensure you have the `@wordpress/hooks` package installed. If not, you can add it to your project using `npm` or `yarn`:
```

Comment:
Do I need to have some min version of npm or yarn?

Reply:
@dakota can you answer this?

Reply:
Nothing has been explicitly documented on the `@wordpress/hooks`, so I think anything that was released in the last 5 to 10 years. Ultimately, it would depend on your own project's needs.